### PR TITLE
feat(typeahead): add `appendElementToId`

### PR DESCRIPTION
**Pull request by wesleycho**
- Add appending popup to specific id

This supercedes #4231

### CODE REVIEW

**Comment by wesleycho** [comment]
Will have doc update shortly

File: src/typeahead/typeahead.js [hunk 1]
```diff
@@ -435,6 +437,8 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position'])
 
         if (appendToBody) {
           $document.find('body').append($popup);
+        } else if (appendToElementId !== false) {
```
Comment (Foxandxss) [inline src/typeahead/typeahead.js]: !== false is needed in here? Any case where it could be falsy but acceptable?

Comment (wesleycho) [inline src/typeahead/typeahead.js]: Given the above setting of the variable to `false` by default, I think this is a fine check.

**Comment by wesleycho** [comment]
Greenlit by @Foxandxss

**Pull request closed without merging**